hyper-parameter-optimization.ipynb

```diff
@@ -0,0 +1,180 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Get resources"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import coiled\n",
+    "import dask.distributed"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "cluster = coiled.Cluster(\n",
+    "    n_workers=20, \n",
+    "    configuration=\"coiled/default\", \n",
+    ")\n",
+    "client = dask.distributed.Client(cluster)\n",
+    "\n",
+    "client"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Get and pre-process data"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import dask.dataframe as dd\n",
+    "df = dd.read_csv(\n",
+    "    \"s3://nyc-tlc/trip data/yellow_tripdata_2019-*.csv\", \n",
+    "    parse_dates=[\"tpep_pickup_datetime\", \"tpep_dropoff_datetime\"],\n",
+    "    dtype={\n",
+    "        \"VendorID\": \"UInt8\",\n",
+    "        \"passenger_count\": \"UInt8\",\n",
+    "        \"RatecodeID\": \"UInt8\",\n",
+    "        \"store_and_fwd_flag\": \"category\",\n",
+    "        \"PULocationID\": \"UInt16\",\n",
+    "        \"DOLocationID\": \"UInt16\",    \n",
+    "        \"payment_type\": \"UInt8\",\n",
+    "    },\n",
+    "    blocksize=\"16 MiB\",\n",
+    ")\n",
+    "\n",
+    "data = df[[\"passenger_count\", \"trip_distance\", \"RatecodeID\", \"payment_type\", \"fare_amount\"]]\n",
+    "data = data.fillna(0)\n",
+    "\n",
+    "labels = (df.tip_amount / df.fare_amount) > 0.25\n",
+    "labels = labels.fillna(False)\n",
+    "\n",
+    "from dask_ml.model_selection import train_test_split\n",
+    "\n",
+    "X_train, X_test, y_train, y_test = train_test_split(data, labels, shuffle=True)\n",
+    "X_train, X_test, y_train, y_test = dask.persist(X_train, X_test, y_train, y_test)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Train model"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from sklearn.linear_model import SGDClassifier\n",
```

Comment:
Yeah, I think I chose SGDClassifier just because it was simple.  These are all black boxes to me, so I chose the simplest black box about which I could find the most examples :)